cmd/chart/upgrade.go

```diff
@@ -0,0 +1,126 @@
+package chart
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/alexellis/arkade/pkg/helm"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/spf13/cobra"
+)
+
+func MakeUpgrade() *cobra.Command {
+	var command = &cobra.Command{
+		Use:   "upgrade",
+		Short: "Upgrade all images in a values.yaml file to the latest version",
+		Long: `Upgrade all images in a values.yaml file to the latest version.
+Container images must be specified at the top level, or one level down in the 
+"image: " or "component.image: " field in a values.yaml file.
+
+Returns exit code zero if all images were found on the remote registry.
+
+Otherwise, it returns a non-zero exit code and the updated values.yaml file.`,
+		Example: `arkade upgrade -f ./chart/values.yaml
+  arkade upgrade --verbose -f ./chart/values.yaml`,
+		SilenceUsage: true,
+	}
+
+	command.Flags().StringP("file", "f", "", "Path to values.yaml file")
+	command.Flags().BoolP("verbose", "v", false, "Verbose output")
+	command.Flags().IntP("depth", "d", 3, "how many levels deep into the YAML structure to walk looking for image: tags")
+
+	command.PreRunE = func(cmd *cobra.Command, args []string) error {
+		_, err := cmd.Flags().GetInt("depth")
+		if err != nil {
+			return fmt.Errorf("error with --depth usage: %s", err)
+		}
+		return nil
+	}
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		file, err := cmd.Flags().GetString("file")
+		if err != nil {
+			return fmt.Errorf("invalid value for flag --file")
+		}
+
+		verbose, _ := cmd.Flags().GetBool("verbose")
+		depth, _ := cmd.Flags().GetInt("depth")
+
+		if len(file) == 0 {
+			return fmt.Errorf("flag --file is required")
+		}
+
+		if ext := path.Ext(file); ext != ".yaml" && ext != ".yml" {
+			return fmt.Errorf("--file must be a YAML file")
+		}
+
+		if verbose {
+			fmt.Printf("Verifying images in: %s\n", file)
+		}
+
+		values, err := helm.Load(file)
+		if err != nil {
+			return err
+		}
+
+		filtered := helm.FilterImagesUptoDepth(values, depth)
+		if len(filtered) == 0 {
+			return fmt.Errorf("no images found in %s", file)
+		}
+
+		if verbose {
+			if len(filtered) > 0 {
+				fmt.Printf("Found %d images\n", len(filtered))
+			}
+		}
+		hasUpdate := false
+		for k, v := range filtered {
+			if verbose {
+				fmt.Printf("> [%s] %s\n", k, v)
+			}
+
+			imageName, tag := splitImageName(k)
+			ref, err := crane.ListTags(imageName)
```

Comment:
Is any pagination required, or does this call return the latest images available?